lib/rucio/tests/test_tpc.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Mayank Sharma <mayank.sharma@cern.ch>, 2021
+
+import time
+import datetime
+import pytest
+import hashlib
+import re
+
+from rucio.core.rule import add_rule
+from rucio.core.transfer import get_transfer_requests_and_source_replicas
+from rucio.common.utils import generate_uuid
+from rucio.daemons.judge.evaluator import re_evaluator
+from rucio.daemons.conveyor import submitter, poller, finisher
+from rucio.client.rseclient import RSEClient
+from rucio.client.ruleclient import RuleClient
+from rucio.common.utils import run_cmd_process
+
+
+@pytest.fixture
+def did_factory(vo, test_scope):
+    from rucio.tests.temp_factories import TemporaryDidFactory
+
+    with TemporaryDidFactory(vo=vo, default_scope=test_scope) as factory:
+        yield factory
+
+
+@pytest.fixture
+def rse_client():
+    return RSEClient()
+
+
+@pytest.fixture
+def rule_client():
+    return RuleClient()
+
+
+def check_url(pfn, hostname, path):
+    assert hostname in pfn
+    assert path in pfn
+
+
+def list_fts_transfer(timeout=30):
+    time_start = time.time()
+    running_time = 0
+    request_id = None
+    request_status = None
+    while running_time < timeout:
+        rcode, out = run_cmd_process("/usr/bin/python2 /usr/bin/fts-rest-transfer-list -v -s https://fts:8446", timeout=10)
+        if "Request ID" in out:
+            request_id = re.search("Request ID: (.*)", out).group(1)
+            request_status = re.search("Status: (.*)", out).group(1)
+            break
+        time_now = time.time()
+        running_time = int(time_now - time_start)
+    return request_id, request_status
+
+
+def poll_fts_transfer_status(request_id, timeout=30):
+    rcode, out = run_cmd_process(f"/usr/bin/python2 /usr/bin/fts-rest-transfer-status -v -s https://fts:8446 {request_id}", timeout=timeout)
+    transfer_status = None
+    if rcode == 0:
+        transfer_status = re.search("Status: (.*)", out).group(1)
+    return transfer_status
+
+
+@pytest.mark.dirty(reason="Creates artifact /tmp/test_tpc.artifact in dev_rucio_1 container")
```

Comment:
should mark it noparallel, multiple submiters cannot run in parallel because the workload is partitioned by the number of submitters. Otherwise, tests will fail randomly. See conveyor_submitter tests for the decorator to add.

Reply:
Gotcha... and done ;)